src/wallets/wallets.js

```diff
@@ -314,27 +344,41 @@ function sameSender(sender1, sender2) {
   return JSON.stringify(sender1.publicKey) == JSON.stringify(sender2.publicKey);
 }
 
+async function signSet(signer, senderAddressObj, txnarray) {
+  const senderAddressObj = algosdk.decodeAddress(info.address);
+  const toSign = txnarray.filter((txn) =>
+      sameSender(txn["from"], senderAddressObj),
+    );
+    const signed = await signer.signTransaction(
+      toSign.map((txn) => txn.toByte()),
+    );
+    let res = [];
+    let sIndex = 0;
+    for (const [index, txn] of txnarray.entries()) {
+      if (sameSender(txn["from"], senderAddressObj)) {
+        res.push(signed[sIndex]);
+        sIndex++;
+      } else {
+        res.push(null);
+      }
+    }
+    return res;
+}
+
 export async function signGroup(info, txnarray) {
   const senderAddressObj = algosdk.decodeAddress(info.address);
   switch (activeWallet.type) {
+      const signedTxns = await signSet(window.exodus.algorand, senderAddressObj, txnarray)
```

Comment:
@ElianaTroper lexical error here, should the code beginning on line 371 be wrapped in a case statement for exodus?